apps/studio/src/lib/db/clients/surrealdb.ts

```diff
@@ -0,0 +1,506 @@
+        return Object.keys(nsInfo.db);
+      }
+      
+      return [];
+    } catch (error) {
+      log.error('Failed to list databases:', error);
+      return [];
+    }
+  }
+
+  async getQuerySelectTop(table: string, limit: number, _schema?: string): Promise<string> {
+    return `SELECT * FROM ${table} LIMIT ${limit}`;
+  }
+
+  async getTableProperties(table: string, _schema?: string): Promise<TableProperties> {
+    const length = await this.getTableLength(table);
+    const indexes = await this.listTableIndexes(table);
+    const triggers = await this.listTableTriggers(table);
+    const relations = await this.getTableKeys(table);
+    
+    return {
+      size: length,
+      indexes,
+      relations,
+      triggers,
+      partitions: []
+    };
+  }
+
+  listMaterializedViews(_filter?: FilterOptions): Promise<TableOrView[]> {
+    return Promise.resolve([]);
+  }
+
+  async getPrimaryKey(_table: string, _schema?: string): Promise<string | null> {
+    // SurrealDB uses 'id' as the primary key by default
+    return 'id';
+  }
+
+  async getPrimaryKeys(_table: string, _schema?: string): Promise<PrimaryKeyColumn[]> {
+    return [{
+      columnName: 'id',
+      position: 1
+    }];
+  }
+
+  async getTableLength(table: string, _schema?: string): Promise<number> {
+    try {
+      const result = await this.db.query(`SELECT count() FROM ${table} GROUP ALL`);
+      return result[0]?.result?.[0]?.count || 0;
+    } catch (error) {
+      log.error('Failed to get table length:', error);
+      return 0;
+    }
+  }
+
+  async selectTop(table: string, offset: number, limit: number, orderBy: OrderBy[], filters: string | TableFilter[], _schema?: string, selects?: string[]): Promise<TableResult> {
+    const sql = await this.selectTopSql(table, offset, limit, orderBy, filters, _schema, selects);
+    const results = await this.executeQuery(sql);
+    const result = results[0];
+    
+    if (!result) {
+      return { result: [], fields: [] };
+    }
+
+    const fields = result.fields.map(f => this.parseTableColumn({ name: f.name, type: 'unknown' }));
+    return { result: result.rows, fields };
+  }
+
+  async selectTopSql(table: string, offset: number, limit: number, orderBy: OrderBy[], filters: string | TableFilter[], _schema?: string, selects?: string[]): Promise<string> {
+    let query = `SELECT ${selects?.join(', ') || '*'} FROM ${table}`;
+    
+    // Add filters
+    if (filters && filters.length > 0) {
+      if (typeof filters === 'string') {
+        query += ` WHERE ${filters}`;
+      } else {
+        const filterClauses = filters.map(filter => 
+          `${filter.field} ${filter.type === '=' ? '==' : filter.type} ${JSON.stringify(filter.value)}`
+        );
+        query += ` WHERE ${filterClauses.join(' AND ')}`;
+      }
+    }
+    
+    // Add ordering
+    if (orderBy && orderBy.length > 0) {
+      const orderClauses = orderBy.map(order => 
+        `${order.field} ${order.dir?.toUpperCase() || 'ASC'}`
```

Comment:
Same here with order.field.

Look into how we do escaping for other db clients (using a DialectData), and do something similar.